packages/scalable-data-sync/src/bloom.ts

```diff
@@ -0,0 +1,59 @@
+import { getMOverNBitsForK } from "./probabilities.js";
+
+export interface BloomFilterOptions {
+  // The expected maximum number of elements for which this BloomFilter is sized.
+  capacity: number;
+
+  // The desired false-positive rate (between 0 and 1).
+  errorRate: number;
+
+  // (Optional) The exact number of hash functions, if the user wants to override the automatic calculation.
+  kHashes?: number;
+
+  // (Optional) Force a specific number of bits per element instead of using a table or optimal formula.
+  forceNBitsPerElem?: number;
+}
+
+export abstract class BloomFilter {
```

Comment:
would be good to have some ts-doc here 🙏

Reply:
added!